### PR TITLE
170 step 1 dipto terraform initialization with lambda

### PR DESCRIPTION
**Pull request by D-sourav-M044**
Fixes #170 

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [X] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
A new lambda function has been created with terraform.

## What was done
A new lambda with source zip code has been initialized with terraform. Iam role for the terraform also has been created.

## How to test
After doing terraform apply, need to check whether tha lambda has been created or not.
- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: Terraform/buckets.tf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+resource "aws_s3_bucket" "myBucket" {
+  bucket = "dipto-terraform-bucket"
```
[the remark this answers is not in the document]
Comment (D-sourav-M044) [reply Terraform/buckets.tf]: Bucket name is now taken from environment variable. [cd24e37](https://github.com/CraftsmenLtd/coe-batch-q322/commit/cd24e37668411c8bebe239e1af0ac2d225ae1dc8)

File: Terraform/dynamo.tf [hunk 2]
```diff
@@ -0,0 +1,52 @@
+resource "aws_dynamodb_table" "my-dynamodb-table" {
+  name           = "dipto-terraform-table"
```
[the remark this answers is not in the document]
Comment (D-sourav-M044) [reply Terraform/dynamo.tf]: Table name is now taken from environment variable. [cd24e37](https://github.com/CraftsmenLtd/coe-batch-q322/commit/cd24e37668411c8bebe239e1af0ac2d225ae1dc8)

File: Terraform/lambda.tf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  runtime          = "nodejs14.x"
+  timeout          = 600
+  environment {
+    variables = {
+      BucketName        = "dipto-terraform-bucket",
+      TableName         = "dipto-terraform-table",
+      SnsArn            = aws_sns_topic.mySns.arn,
+      WebHookUrl        = "https://discord.com/api/webhooks/1004259840325058620/MfNaLLKCfMVzbfUgpFAqS3olMzwBzTTGWV3QFOePArjLN5pw6Z2-eoYBZ6OjhmnYZ-Yi",
+      customerUserTable = "dipto-terraform-client-user-table",
+      customerUserIndex = "dipto-terraform-client-user-index"
```
[the remark this answers is not in the document]
Comment (D-sourav-M044) [reply Terraform/lambda.tf]: These are now taken from environment variable. [cd24e37](https://github.com/CraftsmenLtd/coe-batch-q322/commit/cd24e37668411c8bebe239e1af0ac2d225ae1dc8)

File: Terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+  backend "s3" {
+    bucket = "sourav-craftsmen"
```
[the remark this answers is not in the document]
Comment (D-sourav-M044) [reply Terraform/main.tf]: Backend properties are now taken from environment variable. [cd24e37](https://github.com/CraftsmenLtd/coe-batch-q322/commit/cd24e37668411c8bebe239e1af0ac2d225ae1dc8)

File: Terraform/sqs.tf [hunk 5]
```diff
@@ -0,0 +1,29 @@
+resource "aws_sqs_queue" "sqs_queue" {
+  name                       = "dipto-terraform-sqs"
```
[the remark this answers is not in the document]
Comment (D-sourav-M044) [reply Terraform/sqs.tf]: Sqs name is now taken from environment variable. [cd24e37](https://github.com/CraftsmenLtd/coe-batch-q322/commit/cd24e37668411c8bebe239e1af0ac2d225ae1dc8)

**Review by mhazad — APPROVED** [review]
LGTM

**Pull request merged**